src/wasm/wasm-validator.cpp

```diff
@@ -2853,6 +2863,10 @@ static void validateTables(Module& module, ValidationInfo& info) {
                         segment->offset,
                         "table segment offset should be reasonable");
       validator.validate(segment->offset);
+    } else {
```

Comment:
Maybe I'm not up-to-date with multiple tables stuff... In which case an element segment does not have a table? And why that shouldn't have an offset?

Reply:
I think it might be the difference between active and passive segments, but I don't remember if that's the term for tables (and not just memories). @tlively and @martianboy would know better.

Reply:
Right, it's the same for memories and tables. Passive segments don't belong to a table on instantiation and hence have no offset either. They can be copied to a table using `table.init` instruction which is defined in [reference-types proposal](https://webassembly.github.io/reference-types/core/syntax/instructions.html#syntax-instr-table).

Reply:
Thanks. Do we have passive `elem` segment or `table.init` implemented in binaryen now? I searched with `passive` in test/ directory but that only turns up passive data segments.

Reply:
@aheejin table instructions aren't implemented yet, but we do have passive element segment support. You can take a look at this test: `test/multi-table.wast`.

Reply:
Thanks. So memory passive elements have `passive` keyword but passive data segments don't?

Reply:
No, there is no `passive` keyword. If a data segment has an offset then it's active, otherwise it's considered passive.

Reply:
Where do we have those text format definitions? The spec for [element segments](https://webassembly.github.io/reference-types/core/text/modules.html#element-segments) and [data segments](https://webassembly.github.io/reference-types/core/text/modules.html#data-segments) both have 'passive mode', but neither says it should be written in the text format.

Reply:
Not sure if I understand the question, are you asking whether there should be a keyword specifier? As per spec, if offset and an optional table index are present then it's active, if not it's passive. If `declare` is present in case of a passive element segment, it's considered a declarative segment.

Reply:
No what I'm asking is, 

> No, there is no `passive` keyword. If a data segment has an offset then it's active, otherwise it's considered passive.

You said there is a `passive` keyword in passive element segments but not in passive data segments. But the spec for two segments I linked above are identical.

Reply:
Oh sorry I think you misunderstood me. I was saying there is no `passive` keyword in either case. The spec is very similar in both cases.

Reply:
Hmm I see. I was confused because there are binaryen tests with `(data passive ...)` syntax, like this. https://github.com/WebAssembly/binaryen/blob/9b38e4394034d6f25f54fc471bb9ac8812815331/test/spec/bulk-memory.wast#L4

The `data passive` syntax appears in the [explainer](https://github.com/WebAssembly/bulk-memory-operations/blob/master/proposals/bulk-memory-operations/Overview.md#passive-segment-initialization-example) of the bulk memory proposal, but I guess that's not part of the final spec? The [spec repo](https://github.com/WebAssembly/spec) does not contain tests with that syntax either. I guess we can remove it? cc @tlively

Reply:
You're right, the parser seems to be outdated for data segments. I've also been keeping the "old-syntax" for tables and element segments, and I don't know if it's necessary.

Reply:
In general I don't think we need to support text files outside of our test suite for stuff like this that is not in the upstream spec. Removing old syntax forms and updating our tests sounds good.

Reply:
I made two PRs (#3753 and #3757) to remove old table & elem syntax as well as the `passive` keyword for data segments. @tlively @aheejin